src/index.ts

```diff
@@ -66,24 +76,70 @@ function sendRequest(parameters: any) {
         });
 }
 
-function getCleanJsonData(json: object): object {
-    const jsonString = JSON.stringify(json)
-        // I have no idea why there are so many '\t's '\r's and '\n's in the response
-        // text (specifically in the trans_lang value), but I'm getting rid of them:
-        .replace(/(\\t)*(\\r)*(\\n)*/g, '')
-        // Only changing these for consistency, not sure that they're all that useful:
-        .replace(/\"sup_no\"/g, '"homomorphicNumber"')
-        .replace(/\"target_code\"/g, '"targetCode"')
-        // Bringing these in line with the names of the input parameters:
-        .replace(/\"sense\"/g, '"meaning"')
-        .replace(/\"sense_order\"/g, '"meaningOrder"')
-        .replace(/\"pos\"/g, '"partOfSpeech"')
-        .replace(/\"word_grade\"/g, '"vocabularyGrade"')
-        // These are all in a "translation" property, the preceding 'trans_' isn't useful:
-        .replace(/\"trans_lang\"/g, '"language"')
-        .replace(/\"trans_word\"/g, '"word"')
-        .replace(/\"trans_dfn\"/g, '"definition"');
-    return JSON.parse(jsonString);
+function getCleanJsonData(json: any): object {
+    const stack = [[json, null]];
+    const rename = [];
+
+    while (stack.length > 0) {
+        let [elem, key] = stack.pop()!;
+
+        // check key for renaming & value trimming
+        if (key !== null) {
+            const container = elem;
+            elem = container[key];
+
+            // handle the tab and newline characters
+            if (Array.isArray(elem) && elem.length === 1 && typeof elem[0] === 'string') {
+                // this branch can be removed when explicitArray option is used
+                elem[0] = elem[0].trim();
+            } else if (typeof elem === 'string') {
+                elem = elem.trim();
+                container[key] = elem;
+            }
```

Comment:
For some of these `if` statements, do you think it's okay to move them out into separate functions? 
My preference is to have fewer comments by having more functions with descriptive names if it's possible, so in this case doing something like:
```javascript
function handleTabAndNewline(...) {
    if {
        ...
        return
    }
    ...
}
```
The other preference I have is fewer `else` blocks by doing early `return`s in functions like that but that's less important to me than having descriptive function names, I think 😅

Reply:
@Fox-Islam I've added the changes you mentioned to the if statements involving processing the key and value, but I've left the part which adds elements to the stack alone. Would you prefer that if statement to be extracted as well?

Reply:
@omarkmu Ah, wonderful! No, thank you, I think it looks good, happy to merge this!